exporters/trace/gcp_exporter/internal/gcp_exporter.cc

```diff
@@ -153,36 +30,35 @@ std::unique_ptr<google::devtools::cloudtrace::v2::TraceService::Stub> MakeServic
 }
 
 
-GcpExporter::GcpExporter() : GcpExporter(MakeServiceStub(), GenerateRandomTraceId()) {}
+GcpExporter::GcpExporter() : GcpExporter(MakeServiceStub(), getenv(kGCPEnvVar)) {}
 
 
 GcpExporter::GcpExporter(std::unique_ptr<google::devtools::cloudtrace::v2::TraceService::StubInterface> stub,
-                         const trace::TraceId trace_id):
-    trace_service_stub_(std::move(stub)), trace_id_(trace_id) {}
+                         const char* project_id):
+    trace_service_stub_(std::move(stub)), project_id_(project_id) {}
 
 
 /* ############################### EXPORT FUNCTIONS ################################## */
 
 
 std::unique_ptr<sdk::trace::Recordable> GcpExporter::MakeRecordable() noexcept
 {
-    return std::unique_ptr<sdk::trace::Recordable>(new sdk::trace::SpanData);
+    return std::unique_ptr<sdk::trace::Recordable>(new Recordable);
 }
 
 
 sdk::trace::ExportResult GcpExporter::Export(
       const nostd::span<std::unique_ptr<sdk::trace::Recordable>> &spans) noexcept 
 {
-    // Fetch project id
-    const std::string project_id(getenv(kGCPEnvVar));
-
     // Set up gRPC request
     google::devtools::cloudtrace::v2::BatchWriteSpansRequest request;
-    request.set_name(kProjectsPathStr + project_id);
-
-    // Convert recordables to google protobufs
-    ConvertSpans(spans, project_id, trace_id_, &request);
+    request.set_name(kProjectsPathStr + project_id_);
+    for(auto& recordable: spans){
+        auto span = static_cast<Recordable*>(recordable.release());
```

Comment:
I think there may be a memory leak here - auto would default to Recordable* and not std::unique_ptr<Recordable>, and Recordable will be never deleted because it is released but ownership is not claimed.

It would be better to rewrite this as 

auto span = std::unique_ptr<Recordable>(static_cast<Recordable*>(recordable.release()));

Reply:
Done!